src/main/scala/elo/LeaderboardService.scala

```diff
@@ -1,9 +1,28 @@
 package elo
+import scala.util.Try
 
 class LeaderboardService(playerService: PlayerService) {
 
-  def getLeaderboard(numTopPlayers: Int = 3): Seq[Player] = {
-    playerService.findAllPlayers.sortBy(-_.score).take(numTopPlayers)
+  val defaultLeaderboardSize = 3
+
+  def getLeaderboardSize(messageText: String): Option[Int] = {
+    // get position of token after first "leaderboard" invocation
+    val splitCommand = messageText.split(" ")
+    val posNumPlayers = splitCommand
+      .map(x => x.contains("eaderboard"))
+      .indexOf(true) + 1
+    // "leaderboard" is last token in message, return None
+    // if the token after "leaderboard" cannot be cast to int, return None
+    if(posNumPlayers == splitCommand.length) None else{
+      val numPlayersShow = splitCommand(posNumPlayers)
+      Try(numPlayersShow.toInt).toOption
+    }
+  }
+
+  def getLeaderboard(numTopPlayers: Option[Int]): Seq[Player] = {
```

Comment:
You can actually also specify a default value for parameters - in this case:

```scala
def getLeaderboard(numTopPlayers: Int = DefaultLeaderboardSize): Seq[Player] = {...}
```

Reply:
Do you think we should specify a default and use Int instead of passing an Option[Int] to this method?

Reply:
Fixed this in the most recently pushed code